### ISSUE TITLE
Updating from 5.x to 6.x has file conflicts

### ISSUE THREAD

**Issue by valeriujieanu** [geerlingguy/ansible-role-nodejs#61]
Hello,

We used `5.x` until today when we wanted to update to `6.x`. But there are many file conflicts.

Based on my understanding of Ansible, the below RedHat task:
```
- name: Ensure Node.js and npm are installed.
yum: "name=nodejs-{{ nodejs_version[0] }}.* state=present enablerepo='epel,nodesource'"
```
is equivalent to `yum install nodejs-6.*`. Both fail with the same error. `yum update nodejs` also fails, the log is below.

```
root@staging: ~ > yum update nodejs
Resolving Dependencies
--> Running transaction check
---> Package nodejs.x86_64 0:5.12.0-1nodesource.el7.centos will be updated
---> Package nodejs.x86_64 1:6.9.4-2.el7 will be an update
--> Processing Dependency: npm = 1:3.10.10-1.6.9.4.2.el7 for package: 1:nodejs-6.9.4-2.el7.x86_64
--> Processing Dependency: libuv >= 1:1.9.1 for package: 1:nodejs-6.9.4-2.el7.x86_64
--> Processing Dependency: libuv.so.1()(64bit) for package: 1:nodejs-6.9.4-2.el7.x86_64
--> Processing Dependency: libicuuc.so.50()(64bit) for package: 1:nodejs-6.9.4-2.el7.x86_64
--> Processing Dependency: libicui18n.so.50()(64bit) for package: 1:nodejs-6.9.4-2.el7.x86_64
--> Processing Dependency: libicudata.so.50()(64bit) for package: 1:nodejs-6.9.4-2.el7.x86_64
--> Running transaction check
---> Package libicu.x86_64 0:50.1.2-15.el7 will be installed
---> Package libuv.x86_64 1:1.10.2-1.el7 will be installed
---> Package npm.x86_64 1:3.10.10-1.6.9.4.2.el7 will be installed
--> Finished Dependency Resolution

Dependencies Resolved

=====================================================================================================================================================================================================================
 Package                                        Arch                                           Version                                                            Repository                                    Size
=====================================================================================================================================================================================================================
Updating:
 nodejs                                         x86_64                                         1:6.9.4-2.el7                                                      epel                                         4.6 M
Installing for dependencies:
 libicu                                         x86_64                                         50.1.2-15.el7                                                      base                                         6.9 M
 libuv                                          x86_64                                         1:1.10.2-1.el7                                                     epel                                         109 k
 npm                                            x86_64                                         1:3.10.10-1.6.9.4.2.el7                                            epel                                         2.4 M

Transaction Summary
=====================================================================================================================================================================================================================
Install             ( 3 Dependent packages)
Upgrade  1 Package

Total size: 14 M
Is this ok [y/d/N]: y
Downloading packages:
Running transaction check
Running transaction test


Transaction check error:
  file /usr/lib/node_modules/npm/node_modules/node-gyp from install of npm-1:3.10.10-1.6.9.4.2.el7.x86_64 conflicts with file from package nodejs-5.12.0-1nodesource.el7.centos.x86_64
  file /usr/lib/node_modules/npm/doc from install of npm-1:3.10.10-1.6.9.4.2.el7.x86_64 conflicts with file from package nodejs-5.12.0-1nodesource.el7.centos.x86_64
  file /usr/lib/node_modules/npm/html from install of npm-1:3.10.10-1.6.9.4.2.el7.x86_64 conflicts with file from package nodejs-5.12.0-1nodesource.el7.centos.x86_64
  file /usr/lib/node_modules/npm/man from install of npm-1:3.10.10-1.6.9.4.2.el7.x86_64 conflicts with file from package nodejs-5.12.0-1nodesource.el7.centos.x86_64
```
So I may do a PR to check the current version installed, determine whether the requested version is different than the version installed, if yes remove the old binaries, if yes install the new requested version.

What do you think?

Thank you for your work.

**Comment by geerlingguy** [comment]
@valeriujieanu - For most of my roles, I would rather keep it simple and just make sure that the version provided is the version installed. Sometimes in downstream _playbooks_, I'll add additional logic to help users upgrade or downgrade versions of something, but IMO if infrastructure is to be more idempotent/static, you'd want to do one of two things:

  1. Rebuild the server(s) from scratch.
  2. Create an upgrade playbook, or temporarily add a shim task list that will prep the server for the upgrade by removing services that are slated for upgrade.

Unless it's just a couple lines of changes, (e.g. adding a "state=latest" variable or something like that), I'd rather not add the logic to this role directly.

**Comment by valeriujieanu** [comment]
I wanted to implement the "create new infrastructure, provision it, then destroy the old infra" logic for some time now and now I got the opportunity to do so. 

So, if I have the liberty of using `state=latest` (afford to always update to the latest version available without breaking stuff), that's the easy way to go. If not, it's easier to create new instances with the new state and replace the older ones, instead of dealing with the update of every sensible component.

I hope I got it right. Thanks ! 👍
